src/server/worker/datum/create.go

```diff
@@ -0,0 +1,482 @@
+package datum
+
+import (
+	"context"
+	"math"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/gogo/protobuf/types"
+	"github.com/pachyderm/pachyderm/v2/src/client"
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/storage/renew"
+	"github.com/pachyderm/pachyderm/v2/src/internal/task"
+	"github.com/pachyderm/pachyderm/v2/src/pfs"
+	"github.com/pachyderm/pachyderm/v2/src/pps"
+	"github.com/pachyderm/pachyderm/v2/src/server/worker/common"
+	"golang.org/x/sync/errgroup"
+)
+
+func Create(pachClient *client.APIClient, taskDoer task.Doer, input *pps.Input) (string, error) {
+	switch {
+	case input.Pfs != nil:
+		return createPFS(pachClient, taskDoer, input.Pfs)
+	case input.Union != nil:
+		return createUnion(pachClient, taskDoer, input.Union)
+	case input.Cross != nil:
+		return createCross(pachClient, taskDoer, input.Cross)
+	case input.Join != nil:
+		return createJoin(pachClient, taskDoer, input.Join)
+	case input.Group != nil:
+		return createGroup(pachClient, taskDoer, input.Group)
+	case input.Cron != nil:
+		return createCron(pachClient, taskDoer, input.Cron)
+	default:
+		return "", errors.Errorf("unrecognized input type: %v", input)
+	}
+}
+
+func createPFS(pachClient *client.APIClient, taskDoer task.Doer, input *pps.PFSInput) (string, error) {
+	if input.Commit == "" {
+		// this can happen if a pipeline with multiple inputs has been triggered
+		// before all commits have inputs
```

Comment:
This might be a vestige from prior to global ids. This comment and functionality were pulled directly from the existing pfs iterator. What you said makes sense to me, so I can remove this if you sign off on it.

Reply:
Oh sorry, I missed that it was in the old version. I'm comfortable with removing it